test-data/unit/check-redefine2.test

```diff
@@ -0,0 +1,1076 @@
+-- Test cases for the redefinition of variable with a different type (new version).
+
+[case testNewRedefineLocalWithDifferentType]
+# flags: --allow-redefinition-new --local-partial-types
+def f() -> None:
+    x = 0
+    reveal_type(x) # N: Revealed type is "builtins.int"
+    x = ''
+    reveal_type(x) # N: Revealed type is "builtins.str"
+
+[case testNewRedefineConditionalLocalWithDifferentType]
+# flags: --allow-redefinition-new --local-partial-types
+def f() -> None:
+    if int():
+        x = 0
+        reveal_type(x) # N: Revealed type is "builtins.int"
+    else:
+        x = ''
+        reveal_type(x) # N: Revealed type is "builtins.str"
```

Comment:
Maybe my missed them, but I think it is important to add tests with deeper (like 2-level and maybe even 3-level) nesting of `if` statements, checking the revealed type on exiting each nesting level.